includes/class-wc-amazon-payments-advanced-utils.php

```diff
@@ -54,15 +54,23 @@ public static function get_non_required_fields() {
 	 */
 	public static function get_non_required_fields_per_country() {
 
-		$non_required_fields = array(
+		$mapped_fields_per_country = array(
 			'JP' => array(
-				'billing_city',
-				'shipping_city',
-				'billing-city',
-				'shipping-city',
+				'city',
 			),
 		);
 
-		return apply_filters( 'woocommerce_amazon_pa_non_required_fields_per_country', $non_required_fields );
+		$non_required_fields = array();
+
+		foreach ( $mapped_fields_per_country as $country => $fields ) {
+			foreach ( $fields as $field ) {
+				$non_required_fields[ $country ][] = 'billing_' . $field;
+				$non_required_fields[ $country ][] = 'billing-' . $field;
+				$non_required_fields[ $country ][] = 'shipping_' . $field;
+				$non_required_fields[ $country ][] = 'shipping-' . $field;
+			}
```

Comment:
Again here, why we have two billing and two shipping?

Reply:
It's billing_city for legacy checkout and billing-city for blocks checkout